server/src/main/java/org/opensearch/common/cache/tier/DiskCachingTier.java

```diff
@@ -0,0 +1,18 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.cache.tier;
+
+/**
+ * This is specific to disk caching tier and can be used to add methods which are specific to disk tier.
+ * @param <K> Type of key
+ * @param <V> Type of value
+ */
+public interface DiskCachingTier<K, V> extends CachingTier<K, V> {
```

Comment:
DiskCachingTier looks limiting name wise to me. Is it better to call this OffHeapCachingTier (might extend to remote store in future) for generalization and correctness (even disk tier consumes native memory for reading files)? Disk and Remote can be implementations of OffHeap tier?

Reply:
We can OffHeapCaching tier for disk/off heap use cases, make sense. But I think we should keep RemoteCachingTier as a separate interface as its implementation might vary from onheap/disk ones.

Reply:
> But I think we should keep RemoteCachingTier as a separate interface as its implementation might vary from onheap/disk ones.

I agree with that. Maybe I was not completely clear, but Disk and Remote can both be extensions of OffHeap interface? I am assuming both Disk and Remote can have `OffHeap commonalities` in spite of their many differences.

Reply:
Okay. 
So you mean to have a ```OffHeapCachingTier```, and then ```DiskCachingTier``` extends ```OffHeapCachingTier```?
Similarly below? Though will remote tier later when needed.
```RemoteCachingTier extends OffHeapCachingTier``` ?